### PR TITLE
Graphpocalypse

### PR DESCRIPTION
**Pull request by fommil**
- Google Summer of Code 2016 with Andrey Sugak http://ensime.org/contributing/gsoc2016/#the-graphpocalypse
- Recent Talk at Scalasphere http://ensime.org/talks/scalasphere17/
- [x] release `2.0.0-M1` before merging

### CODE REVIEW

**Comment by fommil** [comment]
looks like github isn't going to let us test CI on this branch, it needs to be mergeable before it builds the incoming merge branch.

**Comment by fommil** [comment]
we're still ignoring a bunch of tests that we shouldn't be ignoring.

**Comment by fommil** [comment]
Windows failure:

```
[info] - should remove classfiles that have been deleted *** FAILED ***
[info]   (1,93291) was not equal to (1,0) (SearchServiceSpec.scala:69)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.MatchersHelper$.indicateFailure(MatchersHelper.scala:340)
[info]   at org.scalatest.Matchers$AnyShouldWrapper.shouldBe(Matchers.scala:6864)
[info]   at org.ensime.indexer.SearchServiceSpec.$anonfun$new$11(SearchServiceSpec.scala:69)
[info]   at org.ensime.fixture.SharedSearchServiceFixture.withSearchService(SearchServiceFixture.scala:49)
[info]   at org.ensime.fixture.SharedSearchServiceFixture.withSearchService$(SearchServiceFixture.scala:47)
[info]   at org.ensime.indexer.SearchServiceSpec.withSearchService(SearchServiceSpec.scala:18)
[info]   at org.ensime.indexer.SearchServiceSpec.$anonfun$new$10(SearchServiceSpec.scala:61)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
```

**Comment by gosubpl** [comment]
Cool, I'd love to help out :) Let me check out this branch and try the tests. Would be great to get some guidance on how to run cached tests a'la drone.yml as when I run them standard way, they fail most of the time due time-outs.

**Comment by fommil** [comment]
don't look in drone for windows, look in `appveyor.yml` for the windows commands

**Comment by gosubpl** [comment]
got it running on `GNU\Linux`, will move to `Windows` soon to investigate the test failure you've mentioned above. How about the `OrientDB` performance on `Windows` ? Last time we spoke, this was the main issue. Has it improved?

**Comment by fommil** [comment]
It's not great.

**Comment by fommil** [comment]
seems like a logic bug, I'm able to reproduce this failure in my local VirtualBox windows image. The windows performance is also very concerning.

**Comment by fommil** [comment]
I've got it down to `ReverseLookupSpec` going green but the forked jvm never closes. That leads me to suspect that the `getVertices(Direction.IN, bdf.label)` call returns an `Iterator` that is somehow not cleaning up properly. Could be spawning a new thread or something. I'm investigating daemonising all the things.

**Comment by fommil** [comment]
still locking up. I have a threaddump, but nothing obvious in there.

**Comment by gosubpl** [comment]
I have tested yesterday evening on `Windows 7` with both `2.11.8` and `2.12.1` with:
```
sbt ++2.11.8 "it:testOnly *SearchServiceSpec"
sbt ++2.12.1 "it:testOnly *SearchServiceSpec"
```
and the test `should remove classfiles that have been deleted` has _passed_ for both cases.

However, running just `it:test` is a disaster due to general `OrientDB` slowness...

I think the suggestion to replace it with something else is the right way to go. I am open for suggestions - could try coding this.

**Comment by fommil** [comment]
@gosubpl remember to run `sbt ensime/assembly` and then `cd testing\cache && sbt ++%SCALA_VERSION% ensimeConfig ensimeServerIndex` before running the integration tests. I think perhaps `SearchService` doesn't need this.

**Comment by gosubpl** [comment]
That's what I did (as described in the `appveyor.xml` file).

File: core/src/it/scala/org/ensime/core/RichPresentationCompilerSpec.scala [hunk 1]
```diff
@@ -52,7 +52,7 @@ class RichPresentationCompilerThatNeedsJavaLibsSpec extends EnsimeSpec
 }
 
 class RichPresentationCompilerSpec extends EnsimeSpec
-    with IsolatedRichPresentationCompilerFixture
+    with SharedRichPresentationCompilerFixture
```
Comment (fommil) [inline core/src/it/scala/org/ensime/core/RichPresentationCompilerSpec.scala]: we should perhaps go back to `Isolated` fixtures where possible. This was a performance hack that we may no longer need.

**Comment by iravid** [comment]
🎉🎉🎉🎉

________________________________

**Pull request merged**